src/pybind/mgr/cephadm/module.py

```diff
@@ -3195,18 +3196,44 @@ def set_custom_prometheus_alerts(self, alerts_file: str) -> str:
 
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
+        try:
+            if url.startswith("http://") or url.startswith("https://"):
+                return f"Invalid service endpoint '{url}'. It should only be in the format host_ip:port."
+
+            # Add brackets around IPv6 address if it contains a port
```

Comment:
this code can fail in several ways ... we can't check/wrap ipv6 like that. In cephadm we already have examples for checking and wraping ipv6, for this you have to use:

`from ceph.deployment.utils import is_ipv6, unwrap_ipv6`

Anyway, It's not clear why are we wrapping the address here .. urlparse should take care of that!! .. if you want to wrap the address then you have to do it later, once you get the hostname (line 3210) and rebuild the endpoint in the `ip:port` format.

Reply:
Thanks @rkachach , Updated the PR. urlparse works fine with Ipv6 with a port. In our case since we allow just ip without port as well so urlparse did not work correctly in that case, made some adjustments please check